simulator/mips/mips_instr.h

```diff
@@ -439,6 +440,13 @@ class BaseMIPSInstr
         bool is_store() const { return operation == OUT_I_STORE; }
 
         bool is_nop() const { return instr.raw == 0x0u; }
+
+        bool is_syscall() const { return operation == OUT_R_SYSCALL; }
+
+        bool is_break() const { return operation == OUT_R_BREAK; }
```

Comment:
I am not sure, but `break` and `trap` instructions should behave similarly. Traps are just conditional breaks, as far as I understand the intention. Probably the best idea is to set up `trap` variable by break and use `has_trap` as a stopping condition